src/test/fuzz/txorphan.cpp

```diff
@@ -402,3 +405,467 @@ FUZZ_TARGET(txorphan_protected, .init = initialize_orphanage)
+                    if ((pattern >> tx) & 1) {
+                        block.vtx.emplace_back(txn[tx]);
+                        for (auto& txin : block.vtx.back()->vin) {
+                            spent.insert(txin.prevout);
+                        }
+                    }
+                }
+                std::shuffle(block.vtx.begin(), block.vtx.end(), rng);
+                real->EraseForBlock(block);
+                std::erase_if(sim_announcements, [&](auto& ann) {
+                    for (auto& txin : txn[ann.tx]->vin) {
+                        if (spent.count(txin.prevout)) return true;
+                    }
+                    return false;
+                });
+                break;
+            } else if (command-- == 0) {
+                // AddChildrenToWorkSet
+                auto tx = read_tx_fn();
+                FastRandomContext rand_ctx(rng.rand256());
+                auto added = real->AddChildrenToWorkSet(*txn[tx], rand_ctx);
+                /** Map of all child wtxids, with value whether they already have a reconsiderable
+                    announcement from some peer. */
+                std::map<Wtxid, bool> child_wtxids;
+                for (unsigned child_tx = 0; child_tx < NUM_TX; ++child_tx) {
+                    if (!have_tx_fn(child_tx)) continue;
+                    bool child_of = false;
+                    for (auto& txin : txn[child_tx]->vin) {
+                        if (txin.prevout.hash == txn[tx]->GetHash()) {
+                            child_of = true;
+                            break;
+                        }
+                    }
+                    if (child_of) {
+                        child_wtxids[txn[child_tx]->GetWitnessHash()] = have_reconsiderable_fn(child_tx);
+                    }
+                }
+                for (auto& [wtxid, peer] : added) {
+                    // Wtxid must be a child of tx.
+                    auto child_wtxid_it = child_wtxids.find(wtxid);
+                    assert(child_wtxid_it != child_wtxids.end());
+                    // Announcement must exist.
+                    auto sim_ann_it = find_announce_wtxid_fn(wtxid, peer);
+                    assert(sim_ann_it != sim_announcements.end());
+                    // Announcement must not yet be reconsiderable.
+                    assert(sim_ann_it->reconsider == false);
+                    // Make reconsiderable.
+                    sim_ann_it->reconsider = true;
+                }
+                for (auto& [wtxid, peer] : added) {
+                    // Remove from child_wtxids map, so we can check that only already-reconsiderable
+                    // ones we not present in added.
+                    child_wtxids.erase(wtxid);
+                }
+                // Verify that all child wtxids which did not occur at least once in the AddChildrenToWorkSet result
+                // are wtxid which were already reconsiderable (for some peer).
```

Comment:
in case this seems more straight forward to understand
```suggestion
                // are wtxid which were already reconsiderable for some peer due to a previous AddChildrenToWorkSet().
```

Reply:
reworded, took this as well